processor/src/lib.rs

```diff
@@ -551,6 +554,11 @@ pub trait ProcessState {
     /// Returns a word located at the specified context/address, or None if the address hasn't
     /// been accessed previously.
     fn get_mem_value(&self, ctx: u32, addr: u32) -> Option<Word>;
+
+    /// Returns the entire memory state for the specified execution context at the current clock
+    /// cycle. The state is returned as a vector of (address, value) tuples, and includes addresses
+    /// which have been accessed at least once.
```

Comment:
minor nit: I'd put a line break after the first sentence. This way, the docs will be formatted better. Specifically:
```
/// Returns the entire memory state for the specified execution context at the current clock
/// cycle.
///
/// The state is returned as a vector of (address, value) tuples, and includes addresses which
/// have been accessed at least once.
```